src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -631,4 +643,130 @@ public static void updateConfigErrorMessage(Config result, String key, String ms
+        topicName, getSchemaRegistryClientFromURL(schemaRegistryURL), "value");
+  }
+
+  private static SchemaRegistryClient getSchemaRegistryClientFromURL(
+      final String schemaRegistryURL) {
+    Map<String, String> srConfig = new HashMap<>();
+    srConfig.put("schema.registry.url", schemaRegistryURL);
+    AvroConverterConfig avroConverterConfig = new AvroConverterConfig(srConfig);
+    return new CachedSchemaRegistryClient(
+        avroConverterConfig.getSchemaRegistryUrls(),
+        avroConverterConfig.getMaxSchemasPerSubject(),
+        Collections.singletonList(new AvroSchemaProvider()),
+        srConfig,
+        avroConverterConfig.requestHeaders());
+  }
+
+  @VisibleForTesting
+  public static Map<String, String> getSchemaFromSchemaRegistryClient(
+      final String topicName, final SchemaRegistryClient schemaRegistry, final String type) {
+    String subjectName = topicName + "-" + type;
+    SchemaMetadata schemaMetadata;
+    try {
+      schemaMetadata = schemaRegistry.getLatestSchemaMetadata(subjectName);
+    } catch (Exception e) {
+      throw SnowflakeErrors.ERROR_0012.getException();
+    }
+    Map<String, String> schemaMap = new HashMap<>();
+    if (schemaMetadata != null) {
+      AvroSchema schema = new AvroSchema(schemaMetadata.getSchema());
+      for (Schema.Field field : schema.rawSchema().getFields()) {
+        Schema fieldSchema = field.schema();
+        if (!schemaMap.containsKey(field.name())) {
+          switch (fieldSchema.getType()) {
+            case BOOLEAN:
+              schemaMap.put(field.name(), "boolean");
+              break;
+            case BYTES:
+              schemaMap.put(field.name(), "binary");
+              break;
+            case DOUBLE:
+              schemaMap.put(field.name(), "double");
+              break;
+            case FLOAT:
+              schemaMap.put(field.name(), "float");
+              break;
+            case INT:
+              schemaMap.put(field.name(), "int");
+              break;
+            case LONG:
+              schemaMap.put(field.name(), "number");
+              break;
+            case STRING:
+              schemaMap.put(field.name(), "string");
+              break;
+            case ARRAY:
+              schemaMap.put(field.name(), "array");
+              break;
+            default:
+              schemaMap.put(field.name(), "variant");
+          }
+        }
+      }
+    }
+    return schemaMap;
+  }
+
+  /**
+   * From the connector config extract whether the avro value converter is used
+   *
+   * @param connectorConfig
+   * @return whether the avro value converter is used
+   */
+  public static boolean usesAvroValueConverter(final Map<String, String> connectorConfig) {
+    List<String> validAvroConverter = new ArrayList<>();
+    validAvroConverter.add("io.confluent.connect.avro.AvroConverter");
+    validAvroConverter.add("com.snowflake.kafka.connector.records.SnowflakeAvroConverter");
```

Comment:
We will not support this. please remove this. 
For Snowpipe streaming we only support confluent AvroConverter.